### PR TITLE
refactor(libs): Extract `basics` lib

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview
Pull out generic utilities from libs/utils and libs/types that aren't VX-specific into their own library to help avoid dependency cycles.

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/basics/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# basics
+
+An extended TS standard library of generic utility functions and types.
+
+This library should only include utilities that are not specific to VotingWorks
+products, but would be generally applicable to any TypeScript project (similar
+to a library like lodash). In other words, it should be considered as an
+extension of the standard TS libraries. This means it shouldn't have very many
```
Comment (eventualbuddha) [inline libs/basics/README.md:8]: Maybe even mandate that it has no workspace dependencies so that everyone can import it without fear of cycles?

**Review by eventualbuddha — COMMENTED** [review]
LGTM so far.

File: libs/basics/README.md (hunk 1 above)
Comment (jonahkagan) [reply libs/basics/README.md]: Good call, updated

File: .circleci/config.yml [hunk 2]
```diff
@@ -904,6 +926,7 @@ workflows:
       - test-libs-ballot-encoder
       - test-libs-ballot-interpreter-nh
       - test-libs-ballot-interpreter-vx
+      - test-libs-basics
```
Comment (adghayes) [inline .circleci/config.yml:929]: Make sure to have @carolinemodic add this to the required tests on Github after merging.

File: libs/basics/package.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ],
+    "*.+(js|jsx|ts|tsx)": [
+      "eslint --quiet --fix"
+    ],
+    "package.json": [
+      "sort-package-json"
+    ]
+  },
+  "dependencies": {
+    "util": "^0.12.4"
```
Comment (adghayes) [inline libs/basics/package.json:34]: This is only used in `setupTests` AFAICT. Is there any reason we couldn't move it to `devDependencies` it, and make one principle of this repo that there are no `dependencies`?

**Review by adghayes — APPROVED** [review]
@jonahkagan I reviewed what you included in the switch and what is left in `libs/utils` and I agree with your erring on the side of not moving things to `libs/basics`, and didn't notice anything else I think should move. I do think `typedAs` would be best to live in `libs/basics` but I'm approving this PR in case it is easier for you or to do it in a follow-up.

File: libs/basics/package.json (hunk 3 above)
Comment (jonahkagan) [reply libs/basics/package.json:34]: It's also used in result.ts unfortunately

**Pull request merged**